server.go

```diff
@@ -473,24 +476,81 @@ func (srv *Server) ActivateAndServe() error {
 	return &Error{err: "bad listeners"}
 }
 
-// Shutdown shuts down a server. After a call to Shutdown, ListenAndServe and
-// ActivateAndServe will return.
+func (srv *Server) getShutdownTimeout() time.Duration {
+	// if not defined, the shutdownTimeout should be > to the readTimeout where the thread are blocked waiting for msg
+	// UDP is using only readTimeout
+	timeout := srv.getReadTimeout()
+
+	if srv.Listener != nil {
+		// if TCP listener is in use, the read timeout may be extended with the idleTimeout
+		idleTimeout := tcpIdleTimeout
+		if srv.IdleTimeout != nil {
+			idleTimeout = srv.IdleTimeout()
+		}
+		if idleTimeout > timeout {
+			timeout = idleTimeout
+		}
+	}
+
+	// add a little time to ensure we have time to wait the thread are released
+	timeout = timeout + time.Second
+	return timeout
+
+}
+
+// Shutdown shuts down gracefully a server. All in progress queries will be dealt with
+// After a call to Shutdown, ListenAndServe and ActivateAndServe will return when all accepted queries are handled
+// or shutdown timeout occurs
```

Comment:
"Shutdown will gracefully shut down a server. All in progress queries will be processed after which the server will stop."

When will a non-nil error be returned?

Reply:
for now, never - I respected the behavior that was before.
Do we want to return an error if the stop was triggered by timeout ?

NOTE: that is true for ShutdownContext, but not for Shutdown => would be a change of behavior.

Reply:
oops you are right. Before there was an error if the server was not started.
I will fix that to have same behavior as before.